src/scripts/main.js

```diff
@@ -1,3 +1,395 @@
+form.method = 'POST';
+
+document.body.appendChild(form);
+
+// поле для Name
+const nameLabel = document.createElement('label');
+
+nameLabel.textContent = 'Name:';
+
+const nameInput = document.createElement('input');
+
+nameInput.type = 'text';
+
+nameInput.name = 'name';
+nameInput.required = true;
+nameInput.setAttribute('data-qa', 'name');
+
+nameLabel.appendChild(nameInput);
+form.appendChild(nameLabel);
+
+// поле для Position
+const positionLabel = document.createElement('label');
+
+positionLabel.textContent = 'Position:';
+
+const positionInput = document.createElement('input');
+
+positionInput.type = 'text';
+
+positionInput.name = 'position';
+positionInput.required = true;
+positionInput.setAttribute('data-qa', 'position');
+
+positionLabel.appendChild(positionInput);
+form.appendChild(positionLabel);
+
+// поле для Office
+const officeLabel = document.createElement('label');
+
+officeLabel.textContent = 'Office:';
+
+const officeSelect = document.createElement('select');
+
+officeSelect.name = 'office';
+officeSelect.required = true;
+
+officeSelect.setAttribute('data-qa', 'office');
+
+// створюю масив міст
+const offices = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+// додаю кожен офіс до option
+offices.forEach((office) => {
+  const option = document.createElement('option');
+
+  option.textContent = office;
+  officeSelect.appendChild(option);
+});
+
+officeLabel.appendChild(officeSelect);
+form.appendChild(officeLabel);
+
+// поле для Age
+const ageLabel = document.createElement('label');
+
+ageLabel.textContent = 'Age:';
+
+const ageInput = document.createElement('input');
+
+ageInput.type = 'number';
+
+ageInput.name = 'age';
+ageInput.required = true;
+ageInput.min = 0;
+ageInput.setAttribute('data-qa', 'age');
+
+ageLabel.appendChild(ageInput);
+form.appendChild(ageLabel);
+
+// поле для Salary
+const salaryLabel = document.createElement('label');
+
+salaryLabel.textContent = 'Salary:';
+
+const salaryInput = document.createElement('input');
+
+salaryInput.type = 'text';
+
+salaryInput.name = 'salary';
+salaryInput.required = true;
+salaryInput.setAttribute('data-qa', 'salary');
+
+salaryLabel.appendChild(salaryInput);
+form.appendChild(salaryLabel);
+
+// обробляю введення зарплати
+salaryInput.addEventListener('blur', () => {
+  const rawValue = salaryInput.value;
+  // видаляю всі символи окрім чисел і розділових крапок
+  const numericValue = parseFloat(rawValue.replace(/[^0-9.]/g, ''));
+
+  if (!isNaN(numericValue)) {
+    salaryInput.value = numericValue.toLocaleString();
+  } else {
+    salaryInput.value = '';
+  }
+});
+
+// кнопка для відправки форми
+const button = document.createElement('button');
+
+button.type = 'submit';
+button.textContent = 'Save to table';
+
+form.appendChild(button);
+
+// обробка натискання кнопки
+button.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const namee = nameInput.value.trim();
```

Comment:
There is a typo in the variable name `namee`. It should be `name` to maintain consistency with the rest of the code.